spec/requests/topics_controller_spec.rb

```diff
@@ -4841,4 +4841,45 @@ def invite_group(topic, expected_status)
       expect(body["group_name"]).to eq(group.name)
     end
   end
+
+  describe "#set_notifications" do
+    describe "initiated by admin" do
+      it "allows fetches user from params if the request is_api? " do
+        api_key = Fabricate(:api_key, user: admin)
+        post "/t/#{topic.id}/notifications",
+             params: {
+               username: user.username,
+               notification_level: NotificationLevels.topic_levels[:watching],
+             },
+             headers: {
+               HTTP_API_KEY: api_key.key,
+               HTTP_API_USERNAME: admin.username,
+             }
+        expect(TopicUser.find_by(user: user, topic: topic).notification_level).to eq(
+          NotificationLevels.topic_levels[:watching],
+        )
+      end
+    end
+
+    describe "initiated by non-admin" do
+      it "only acts on current_user and ignores `username` param" do
```

Comment:
What happens in the case of API call where `HTTP_API_USERNAME` is a non-admin user? Does it return anything at all? (It may be blocked earlier in the stack.) Would be good to have a test case for this as well.

Reply:
Added. It actually updates the level for that user which is the correct behavior I guess?